src/Matrix.cpp

```diff
@@ -0,0 +1,181 @@
+#include "Matrix.h"
+#include "ImageWrapper.h"
+
+
+//should probably convert this to a template
+
+matrix::matrix() : size(0), p_columns(0), p_rows(0), m_rows(0), m_columns(0), m_input_pixel_vector(0),
+m_size(0), m_ptr_double(nullptr) {}
+
+matrix::matrix(const uint8_t rows, const uint8_t columns, const std::vector<double> &greyscalevalues)
```

Comment:
You are passing greyscale values as a const reference which is good. No copy is made here but see next comment V